syntax="proto2";
option cc_generic_services = true;
package peleus.plugins;

message HttpVisitorServiceRequest {
    required string conn_name = 1;
    required string uri = 2;
    required bool post = 3;
    optional string post_data = 4;
};

message HttpVisitorServiceResponse {
    required int32 error_code = 1;
    required string error_text = 2;
    optional string response = 3;
};

service HttpVisitorService {
    rpc query(HttpVisitorServiceRequest) returns (HttpVisitorServiceResponse);
};
